server/svix-server/src/redis/sentinel.rs

```diff
@@ -0,0 +1,53 @@
+use axum::async_trait;
+use redis::{
+    sentinel::{SentinelClient, SentinelNodeConnectionInfo, SentinelServerType},
+    ErrorKind, IntoConnectionInfo, RedisError,
+};
+use tokio::sync::Mutex;
+
+struct LockedSentinelClient(pub(crate) Mutex<SentinelClient>);
```

Comment:
I left a comment on the omniqueue side about favoring mutex over rwlock. Same question, different repo.

Any read-only access anticipated? If so, it might be a nice thing to switch to rwlock.

I guess even while the inner is pub, the outer is not so maybe we can quietly switch later without any impact.

Reply:
Commented on other PR, but I have a pending change in redis-rs that will hopefully eliminate the need for a Mutex here, so stay tuned.